modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_sidebar/CriteriaSidebarSearchBar.es.js

```diff
@@ -0,0 +1,54 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import ClayButton from '../shared/ClayButton.es';
+
+class CriteriaSidebarSearchBar extends Component {
+	static propTypes = {
+		onChange: PropTypes.func,
+		searchValue: PropTypes.string
+	};
+
+	_handleChange = event => {
+		this.props.onChange(event.target.value);
+	}
+
+	_handleClear = event => {
+		event.preventDefault();
+
+		this.props.onChange('');
+	}
+
+	render() {
+		const {searchValue} = this.props;
+
+		return (
+			<div className="input-group">
+				<div className="input-group-item">
+					<input
+						className="form-control input-group-inset input-group-inset-after"
+						onChange={this._handleChange}
+						placeholder={Liferay.Language.get('search')}
+						type="text"
+						value={searchValue}
+					/>
+
+					<div className="input-group-inset-item input-group-inset-item-after">
+						{searchValue ?
```

Comment:
I'm not totally sold on a ternary that just changes the props for the same component since often the conditional indicates its ComponentA or ComponentB. But this also may be the cleanest way to do it, what're your thoughts?

Other ways I can think of is
```jsx
const buttonProps = searchValue ? {/*...*/} : {/*...*/};

<ClayButton {...buttonProps} />
```

or

```jsx
<ClayButton
    iconName={searchValue ? 'times' : 'search'}
    onClick={searchValue && this._handleClear}
    style="unstyled"
 />
```

Reply:
That makes sense. I'll change it to the second approach since it's more clear which values are changing.